platforms/core-configuration/configuration-cache/src/test/kotlin/org/gradle/configurationcache/serialization/codecs/JavaObjectSerializationCodecTest.kt

```diff
@@ -133,30 +131,44 @@ class JavaObjectSerializationCodecTest : AbstractUserTypeCodecTest() {
     }
 
     @Test
-    fun `Externalizable problems link to the Java serialization section`() {
-        val problems = serializationProblemsOf(ExternalizableBean())
-        assertThat(problems.size, equalTo(1))
-        assertThat(
-            problems[0].documentationSection,
-            equalTo(NotYetImplementedJavaSerialization)
-        )
+    fun `can handle Externalizable beans`() {
```

Comment:
🧪 It'd be good to prove `Externalizable` identities are preserved as well

Reply:
I had a question about that: should we ensure it is preserved, or ensure it is *not* preserved? I was assuming/suspecting externalizable objects were not reused, but I was wrong about that.

Reply:
Done!